core/unit_test/TestReducers.hpp

```diff
@@ -319,6 +329,89 @@ struct TestReducers {
       value = value || values(i);
     }
   };
+
+  // get number of teams for TeamPolicy depending on the tested type
+  constexpr static int get_num_teams() {
+    if constexpr (sizeof(Scalar) == 1) {
+      return 126;
+    } else if constexpr (std::is_same_v<Scalar,
+                                        Kokkos::Experimental::bhalf_t>) {
+      return 256;
+    }
+
+    return 1024;
+  }
+
+  static void test_sum_team_policy(int N, SumFunctor f, Scalar reference_sum) {
+#ifdef KOKKOS_ENABLE_OPENACC
+    if constexpr (std::is_same_v<ExecSpace, Kokkos::Experimental::OpenACC> &&
+                  (std::is_same_v<Scalar, size_t> ||
+                   std::is_same_v<Scalar, double>)) {
+      return;  // FIXME_OPENACC
+    }
+#endif
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+    if constexpr (std::is_same_v<ExecSpace,
+                                 Kokkos::Experimental::OpenMPTarget>) {
+      return;  // FIXME_OPENMPTARGET
+    }
+#endif
+#ifdef KOKKOS_ENABLE_SYCL
+    if constexpr (std::is_same_v<ExecSpace, Kokkos::Experimental::SYCL> &&
+                  (std::is_same_v<Scalar, array_reduce<float, 1>> ||
+                   std::is_same_v<Scalar, array_reduce<float, 2>> ||
+                   std::is_same_v<Scalar, array_reduce<float, 3>> ||
+                   std::is_same_v<Scalar, array_reduce<float, 4>> ||
+                   std::is_same_v<Scalar, array_reduce<float, 7>> ||
+                   std::is_same_v<Scalar, int8_t>)) {
+      return;  // FIXME_SYCL
+    }
+#endif
```

Comment:
Hmm.... I only see problems with `arry_reduce<float, 7>` on Intel GPUs.

Reply:
That's right, I have removed the spurious cases (895201b1b9c69c755b7dac9777d6c0b4fa487755). Still not sure what's the root cause though.